cluster-autoscaler/simulator/dynamicresources/snapshot/patchset_test.go

```diff
@@ -0,0 +1,821 @@
+		searchKey      string
+		wantInCurrent  bool
+	}{
+		"SetCurrent_NewKey_InitiallyEmptyPatchSet": {
+			patchLayers: []map[string]*int{},
+			mutatePatchSet: func(ps *patchSet[string, int]) {
+				ps.SetCurrent("a", 1)
+			},
+			searchKey:     "a",
+			wantInCurrent: true,
+		},
+		"SetCurrent_NewKey_OnExistingEmptyLayer": {
+			patchLayers: []map[string]*int{{}},
+			mutatePatchSet: func(ps *patchSet[string, int]) {
+				ps.SetCurrent("a", 1)
+			},
+			searchKey:     "a",
+			wantInCurrent: true,
+		},
+		"SetCurrent_OverwriteKey_InCurrentPatch": {
+			patchLayers: []map[string]*int{{"a": pint(10)}},
+			mutatePatchSet: func(ps *patchSet[string, int]) {
+				ps.SetCurrent("a", 1)
+			},
+			searchKey:     "a",
+			wantInCurrent: true,
+		},
+		"SetCurrent_OverwriteKey_InLowerPatch_AfterFork": {
+			patchLayers: []map[string]*int{{"a": pint(10)}},
+			mutatePatchSet: func(ps *patchSet[string, int]) {
+				ps.Fork()
+				ps.SetCurrent("a", 1)
+			},
+			searchKey:     "a",
+			wantInCurrent: true,
+		},
+		"DeleteCurrent_ExistingKey_InCurrentPatch": {
+			patchLayers: []map[string]*int{{"a": pint(1)}},
+			mutatePatchSet: func(ps *patchSet[string, int]) {
+				ps.DeleteCurrent("a")
+			},
+			searchKey:     "a",
+			wantInCurrent: false,
+		},
+		"DeleteCurrent_ExistingKey_InLowerPatch_AfterFork": {
+			patchLayers: []map[string]*int{{"a": pint(1)}},
+			mutatePatchSet: func(ps *patchSet[string, int]) {
+				ps.Fork()
+				ps.DeleteCurrent("a")
+			},
+			searchKey:     "a",
+			wantInCurrent: false,
+		},
+		"DeleteCurrent_NonExistentKey_OnExistingEmptyLayer": {
+			patchLayers: []map[string]*int{{}},
+			mutatePatchSet: func(ps *patchSet[string, int]) {
+				ps.DeleteCurrent("x")
+			},
+			searchKey:     "x",
+			wantInCurrent: false,
+		},
+		"DeleteCurrent_NonExistentKey_InitiallyEmptyPatchSet": {
+			patchLayers: []map[string]*int{}, // Starts with no layers
+			mutatePatchSet: func(ps *patchSet[string, int]) {
+				ps.DeleteCurrent("x")
+			},
+			searchKey:     "x",
+			wantInCurrent: false,
+		},
+		"InCurrentPatch_KeySetInCurrent": {
+			patchLayers:    []map[string]*int{{"a": pint(1)}},
+			mutatePatchSet: func(ps *patchSet[string, int]) {},
+			searchKey:      "a",
+			wantInCurrent:  true,
+		},
+		"InCurrentPatch_KeySetInLower_NotInCurrent_AfterFork": {
+			patchLayers:    []map[string]*int{{"a": pint(1)}},
+			mutatePatchSet: func(ps *patchSet[string, int]) { ps.Fork() },
+			searchKey:      "a",
+			wantInCurrent:  false,
+		},
+		"InCurrentPatch_KeyDeletedInCurrent": {
+			patchLayers: []map[string]*int{{"a": pint(1)}},
+			mutatePatchSet: func(ps *patchSet[string, int]) {
+				ps.DeleteCurrent("a")
+			},
+			searchKey:     "a",
+			wantInCurrent: false, // Get in current patch won't find it.
+		},
+		"InCurrentPatch_NonExistentKey_InitiallyEmptyPatchSet": {
+			patchLayers:    []map[string]*int{},
+			mutatePatchSet: func(ps *patchSet[string, int]) { /* InCurrentPatch called below will fork */ },
```

Comment:
nit: Is the comment here up-to-date?

Reply:
Ah, I missed the new `Fork()` calls in the `Current`  patchSet methods. This is only for testing, right? IMO seems fine for `SetCurrent` and `DeleteCurrent`, but `InCurrentPatch` being mutating in any way is very surprising. Can't we just answer false if there are no layers?